### ISSUE TITLE
Can't reproduce  the result of Open-RS3, which is  reported in paper?

### ISSUE THREAD

**Issue by xiaozhang23957** [knoveleng/open-rs#12]
Reproducing the Open-RS3 model, and test on the AIME24 dataset resulted in an accuracy of only 26.7; however, the reported accuracy is 46.7. The accuracy reported by the repository for the Open-RS3 model does not align with the result of 46.7 either. Here is our training config:

`# Model arguments
model_name_or_path: ./DeepSeek-R1-Distill-Qwen-1.5B
model_revision: main
torch_dtype: bfloat16
attn_implementation: flash_attention_2

# Data training arguments
dataset_name: ./open-rs
system_prompt: "A conversation between User and Assistant. The user asks a question, and the Assistant solves it. The assistant first thinks about the reasoning process in the mind and then provides the user with the answer, and put your final answer within \\boxed{{}} . The reasoning process and answer are enclosed within <think> </think> and <answer> </answer> tags, respectively, i.e., <think> reasoning process here </think> <answer> answer here </answer>. Note that respond by English, NOT use other languages."

# GRPO trainer config
bf16: true
use_vllm: false
vllm_device: auto
vllm_enforce_eager: true
vllm_gpu_memory_utilization: 0.7
vllm_max_model_len: 4608
do_eval: false
gradient_accumulation_steps: 4
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
hub_model_id: OpenRS-GRPO
hub_strategy: every_save
learning_rate: 1.0e-06
log_completions: true
log_level: info
logging_first_step: true
logging_steps: 1
logging_strategy: steps
lr_scheduler_type: cosine_with_min_lr
lr_scheduler_kwargs:
  min_lr_rate: 0.1
max_prompt_length: 512
max_completion_length: 3584
max_steps: 500
num_generations: 6
num_train_epochs: 1
output_dir: data/OpenRS-GRPO-rs3
overwrite_output_dir: true
per_device_eval_batch_size: 6
per_device_train_batch_size: 6
push_to_hub: false
report_to:
- tensorboard
reward_funcs:
- format
- cosine
reward_weights:
- 1.0
- 2.0
save_strategy: "steps"
save_steps: 50
seed: 42
temperature: 0.7
warmup_ratio: 0.1`

**Comment by QuyAnh2005** [comment]
@undyingjoker  Which checkpoint did you choose?

You can follow[ this issue](https://github.com/knoveleng/open-rs/issues/9) to choose the correct checkpoint.

**Comment by xiaozhang23957** [comment]
> [@undyingjoker](https://github.com/undyingjoker) Which checkpoint did you choose?
> 
> You can follow[ this issue](https://github.com/knoveleng/open-rs/issues/9) to choose the correct checkpoint.
Thank you very much for your answer.
I choose the 50-step checkpoint (DeepSeek-R1-Distill-Qwen-1.5B training on open-rs datasets) for testing.
